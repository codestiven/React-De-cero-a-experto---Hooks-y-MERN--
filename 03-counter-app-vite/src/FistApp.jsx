import React from 'react'
import PropTypes from 'prop-types'; 



const getsaludo = () => {
  return `Hola Mundo ${new Date().getFullYear()}`
}

const FistApp = ( {
  title = "hola mundo" ,
  subtitle = "Soy un subtitulo",
  numero = 5
} ) => {

  return (
    <>
      <h1 data-testid="test-title" >{title}</h1>
      <h2>{subtitle}</h2>
      <h2>Stiven</h2>
      <p> {numero + 1} </p>


    </>
  )
}

FistApp.propTypes = {
  title: PropTypes.string.isRequired,

}

FistApp.defaultProps = {
  title: "No hay titulo",
  numero: 0,
}

export default FistApp
