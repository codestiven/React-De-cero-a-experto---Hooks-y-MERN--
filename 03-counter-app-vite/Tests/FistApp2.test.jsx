import { render , screen } from "@testing-library/react"
import FistApp from "../src/FistApp"
import React from 'react';

describe('Pruevas en <FistApp/>', () => {

    const title = "Hola, soy Goku"



    test('debe de hacer match con el snapshot', () => {

        const {container} = render(<FistApp title="Hola, soy Goku" subtitle="Soy un subtitulo" />)
        // expect(container).toMatchSnapshot()
      
    })
    
    test('debe de mostrar el mensaje "Hola, soy Goku" ', () => {
        render(<FistApp title={title} />);

        expect(screen.getByText(title)).toBeTruthy()

      
    })
    
    test('debe de mostrar el titulo en un h1', () => {
        render(<FistApp title={title} />);

        expect( screen.getByRole('heading',{ level: 1 }).innerHTML ).toContain(title)

    })

    test('debe de mostrar el subtitulo enviado por props', () => {

        render(<FistApp title={title} />);

        expect(screen.getByText("Soy un subtitulo")).toBeTruthy()
      
    })
    
    
    


})
