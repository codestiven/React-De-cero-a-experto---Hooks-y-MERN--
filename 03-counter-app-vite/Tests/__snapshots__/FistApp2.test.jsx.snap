// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruevas en <FistApp/> debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, soy Goku
  </h1>
  <h2>
    Soy un subtitulo
  </h2>
  <h1>
    Stiven
  </h1>
  <p>
     
    6
     
  </p>
</div>
`;
